### PR TITLE
Add reflect Symbol.info and TypeRepr.asSeenFrom

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
We found this missing method because of this use case https://github.com/lampepfl/dotty/issues/11685#issuecomment-1108270577.

**Review by smarter — CHANGES_REQUESTED** [review]
- I think `tpe` as a name is too close to typeRef/termRef while doing something different. I'm not a fan of `info` but at least it's established jargon in Scala.
- If we add this we should also add `asSeenFrom` since most of the time we're interested in the info as seen from a specific prefix and forgetting to use the correct prefix is a common source of bugs.  This was also requested recently: https://github.com/lampepfl/dotty/discussions/14957
- For the purpose of overriding we should provide some convenience method (`Symbol.newMethodOverride` ?) which given a method symbol in a parent takes care of generating a new symbol with the correct info and flags to override it.

File: tests/run-macros/i11685/Macro_1.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    val invalidMethods = methods.flatMap(checkMethod[R])
+    if (invalidMethods.nonEmpty) {
+      report.errorAndAbort(s"Invalid methods: ${invalidMethods.mkString(", ")}")
+    }
+
+    val className = "_Anon"
+    val parents = List(TypeTree.of[Object], TypeTree.of[T])
+
+    def decls(cls: Symbol): List[Symbol] = methods.map { method =>
+      Symbol.newMethod(cls, method.name, method.info, flags = Flags.EmptyFlags /*TODO: method.flags */, privateWithin = method.privateWithin.fold(Symbol.noSymbol)(_.typeSymbol))
```
Comment (smarter) [inline tests/run-macros/i11685/Macro_1.scala:70]: I see you opened https://github.com/lampepfl/dotty/issues/15035 but it's classified as an enhancement when I think it should be required to stabilize Symbol.newClass

Comment (nicolasstucki) [reply tests/run-macros/i11685/Macro_1.scala:70]: I'm working on that one now. I intend to open a separate PR for that one as it is not directly related with the `info`.

**Comment by ScalaWilliam** [comment]
Very keen to see this get into the mainline!

File: tests/run-macros/i11685/Macro_1.scala (hunk 1 above)
Comment (nicolasstucki) [reply tests/run-macros/i11685/Macro_1.scala:70]: See #15041

**Pull request closed without merging**